narwhal/storage/src/header_store.rs

```diff
@@ -36,7 +37,12 @@ impl HeaderStore {
     }
 
     pub fn write(&self, header: &Header) -> Result<(), TypedStoreError> {
-        self.store.insert(&header.digest(), header)
+        fail_point!("narwhal-store-before-write");
+
+        let result = self.store.insert(&header.digest(), header);
+
+        fail_point!("narwhal-store-after-write");
+        result
```

Comment:
You'll have to add `#[allow(clippy::let-and-return)]` here (and elsewhere) to pass clippy warnings in CI, as clippy assumes that since `result` isn't used in the `fail_point` macro, it is an unneccesary intermediate let binding

Reply:
ty!